java-spiffe-core/src/main/java/io/spiffe/svid/x509svid/X509Svid.java

```diff
@@ -0,0 +1,169 @@
+     * @throws X509SvidException if there is an error parsing the given certsFilePath or the privateKeyFilePath
+     */
+    public static X509Svid load(@NonNull final Path certsFilePath, @NonNull final Path privateKeyFilePath) throws X509SvidException {
+        byte[] certsBytes;
+        try {
+            certsBytes = Files.readAllBytes(certsFilePath);
+        } catch (IOException e) {
+            throw new X509SvidException("Cannot read certificate file", e);
+        }
+
+        byte[] privateKeyBytes;
+        try {
+            privateKeyBytes = Files.readAllBytes(privateKeyFilePath);
+        } catch (IOException e) {
+            throw new X509SvidException("Cannot read private key file", e);
+        }
+        return createX509Svid(certsBytes, privateKeyBytes);
+    }
+
+    /**
+     * Parses the X.509 SVID from PEM or DER blocks containing certificate chain and key
+     * bytes. The key must be a PEM or DER block with PKCS#8.
+     *
+     * @param certsBytes      chain of certificates as a byte array
+     * @param privateKeyBytes private key as byte array
+     * @return a {@link X509Svid} parsed from the given certBytes and privateKeyBytes
+     * @throws X509SvidException if the given certsBytes or privateKeyBytes cannot be parsed
+     */
+    public static X509Svid parse(@NonNull final byte[] certsBytes, @NonNull final byte[] privateKeyBytes) throws X509SvidException {
+        return createX509Svid(certsBytes, privateKeyBytes);
+    }
+
+    /**
+     * @return the chain of certificates as an array of {@link X509Certificate}
+     */
+    public X509Certificate[] getChainArray() {
+        return chain.toArray(new X509Certificate[0]);
+    }
+
+    private static X509Svid createX509Svid(final byte[] certsBytes, final byte[] privateKeyBytes) throws X509SvidException {
+
+        List<X509Certificate> x509Certificates;
+        try {
+            x509Certificates = CertificateUtils.generateCertificates(certsBytes);
+        } catch (CertificateParsingException e) {
+            throw new X509SvidException("Certificate could not be parsed from cert bytes", e);
+        }
+
+        PrivateKey privateKey;
+        try {
+            privateKey = CertificateUtils.generatePrivateKey(privateKeyBytes);
+        } catch (InvalidKeySpecException | InvalidKeyException | NoSuchAlgorithmException e) {
+            throw new X509SvidException("Private Key could not be parsed from key bytes", e);
+        }
+
+        SpiffeId spiffeId;
+        try {
+            spiffeId = CertificateUtils.getSpiffeId(x509Certificates.get(0));
+        } catch (CertificateException e) {
+            throw new X509SvidException(e.getMessage(), e);
+        }
+
+        validatePrivateKey(privateKey, x509Certificates);
+        validateLeafCertificate(x509Certificates.get(0));
+
+        if (x509Certificates.size() > 1) {
+            validateSigningCertificates(x509Certificates.subList(1, x509Certificates.size()));
```

Comment:
This allocation of a new list can be avoided by just passing the start and end indexes of the signing certificates to `validateSigningCertificates()`.

Reply:
Removed the allocation of a new list, passing now the entire list.